polkadot/xcm/xcm-executor/src/traits/hrmp.rs

```diff
@@ -0,0 +1,23 @@
+use xcm::v3::{
+	prelude::{XcmError, XcmResult},
+};
+
+pub trait HandleHrmp {
```

Comment:
@JuaniRios Thanks for the quick PR :) Let's first resolve this trait(s). So the question is: "one trait with functions per notification" or "one trait per notification"? From what I understand, @franciscoaguirre prefers the second option, and I also think it's better and much more flexible, even if it is more verbose.

So, my suggestion:

```
/// TODO: add some docs
pub trait HandleHrmpChannelAccepted {
     fn handle(sender: u32, max_message_size: u32, max_capacity: u32) -> XcmResult
}

/// TODO: add some docs
trait HandleHrmpChannelClosing {
     fn handle(initiator: u32, sender: u32, recipient: u32) -> XcmResult
}

/// TODO: add some docs
trait HandleHrmpNewChannelOpenRequest {
     fn handle(sender: u32, max_message_size: u32, max_capacity: u32) -> XcmResult   
}
```

Add tuple implementation for all, so that we can chain multiple implementations:
```
#[impl_trait_for_tuples::impl_for_tuples(30)]
impl HandleHrmpChannelAccepted for Tuple {
...
}
#[impl_trait_for_tuples::impl_for_tuples(30)]
impl HandleHrmpChannelClosing for Tuple {
...
}
#[impl_trait_for_tuples::impl_for_tuples(30)]
impl HandleHrmpNewChannelOpenRequest for Tuple {
...
}
```

Add them to `xcm_executor::Config`:
```
pub trait Config {
   ...

   type HrmpChannelAcceptedHandler: HandleHrmpChannelAccepted;
   type HrmpChannelClosingHandler: HandleHrmpChannelClosing;
   type HrmpNewChannelOpenRequestHandler: HandleHrmpNewChannelOpenRequest;
}
```


So add the end we can configure runtime like this:

```
impl xcm_executor::Config for XcmConfig {
 ...
   type HrmpChannelAcceptedHandler = (
      PolimecFunding,   
      PalletXcmVersionDiscoveryQueuer,      
   );
   type HrmpChannelClosingHandler = ();                        // <- by default, can be hidden by DefaultConfig
   type HrmpNewChannelOpenRequestHandler = ();      // <- by default, can be hidden by DefaultConfig
}
```

So, at the end, you will just need to implement only the trait `HandleHrmpChannelAccepted` for your pallet     PolimecFunding instead of this: https://github.com/Polimec/polimec-node/blob/28c7bf1313f79ab3248933be4f4f5cc41eb94f1c/pallets/funding/src/lib.rs#L1656-L1669

Reply:
I like it!
Will work on it rn

Reply:
@bkontur @franciscoaguirre  should the `()` implementation return an error, or Ok(())?

If I just implement the trait with the tuples macro, then it will return ok with an empty one with this code:
```rust
#[impl_trait_for_tuples::impl_for_tuples(30)]
impl HandleHrmpChannelOpenRequest for Tuple {
	fn handle(sender: u32, max_message_size: u32, max_capacity: u32) -> XcmResult {
		for_tuples!( #( Tuple::handle(sender, max_message_size, max_capacity)?; )* );
		Ok(())
	}
}
```

Reply:
I think there's no need to return an error if it's not implemented, so current behavior looks good